packages/amplify-cli-core/src/amplify-lockfile-dependency-detector/lock-file-types.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * lock file type
+ */
+export enum LockfileType {
+    NPM = 'npm',
+    YARN = 'yarn',
+}
+
+/**
+* package.json type
+*/
+export type PkgJsonType = {
+    name : string,
+    version: string,
+    dependencies: Record<string, string>,
+    devDependencies?: Record<string, string>
```

Comment:
I don't think for amplify we need dev/peer dependencies check.

Reply:
I think we don't need this check for dev dependencies.
But I'm not sure about peer deps. Can you check what happens if you move extensibility helper/cdk from dependencies to peer deps ? I.e. whether override/custom mini project compiles and works as expected or not? If it does then we should check peers.

Reply:
added yarn2 parser since amplify project was working with yarn2

Reply:
I checked with adding helper package to peer dependencies:
<img width="740" alt="Screen Shot 2023-01-18 at 2 52 15 PM" src="https://user-images.githubusercontent.com/39866697/213312765-610450be-9402-40b0-a9b3-e1e76c03e467.png">

amplify build fails with this error:
<img width="1298" alt="Screen Shot 2023-01-18 at 2 53 00 PM" src="https://user-images.githubusercontent.com/39866697/213312865-950458d7-fe60-489c-b7f7-924e2740e14e.png">

Though I have added a check on peer dependencies as well in package.json and  also detector will be able to detect this package as in lock files, this will be added as dependency